File: yoonju/BOJ/Silver/Silver-4/2847. 게임을 만든 동준이.js
const fs = require("fs");

const [N, ...scores] = fs
  .readFileSync("dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map(Number);

let prev = 20001;

const answer = scores.reverse().reduce((acc, cur) => {
  if (prev <= cur) {
    const change_count = cur - prev + 1;
    prev = prev - 1;
    return acc + change_count;
  }
  prev = cur;
  return acc;
}, 0);

console.log(answer);

File: jieun/BOJ/2847. 게임을 만든 동준이.py
N = int(input())
scores = [int(input()) for _ in range(N)]
answer = 0

for i in range(N - 1, 0, -1):
    if scores[i - 1] < scores[i]:   continue
    answer += scores[i - 1] - scores[i] + 1
    scores[i - 1] = scores[i] - 1

print(answer)
